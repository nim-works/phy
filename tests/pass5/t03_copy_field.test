(TypeDefs
  (Int 8)
  (Record 8 8
    (Field 0 (Type 0)))
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 2)
    (List (Type 1))
    (List
      (Block (Params) (Locals) (List)
        (StorageLive (Loc 0))
        (Drop
          (Copy (Field (Loc 0) 0)))
        (Drop
          (Move (Field (Loc 0) 0)))
        (StorageEnd (Loc 0))
        (Return)))))
