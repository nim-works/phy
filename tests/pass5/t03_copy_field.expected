;$sexp
(TypeDefs
  (Int 8)
  (Record 8 8
    (Field 0 (Type 0)))
  (ProcTy (Void))
  (UInt 8))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 2) 8
    (List
      (Block
        (Params (Type 3))
        (Locals)
        (Drop
          (Load (Type 0)
            (Addr
              (Field (Deref (Type 1) (Copy (Param 0))) 0))))
        (Drop
          (Load (Type 0)
            (Addr
              (Field (Deref (Type 1) (Copy (Param 0))) 0))))
        (Return)))))
